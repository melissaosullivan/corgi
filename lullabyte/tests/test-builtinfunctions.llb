void main(){
	int[] one;
	int[] two;
	int[] three;

	sound a;
	pitch b;
	sound c;

	a = |A4, A5, B6, B7|:0.5:100;
	b = B5;
	c = |C2|:5.0:1;

	one = [1];
	two = [1, 2];
	three = [1, 2, 3];

	print(getPitches(a));
	print(getPitches(b));

	print(getDuration(a));
	print(getDuration(c));

	print(getAmplitude(a));
	print(getAmplitude(c));
	
	print(length(one));
	print(length(two));
	print(length(three));
//	print(length([]));

	a = setDuration(a, 2.5+2.5);
	print(getDuration(a));

	a = setAmplitude(a, 45+5);
	print(getAmplitude(a));

	a = setPitches(a, [C4+1, F4+12]);
	print(getPitches(a));

	a = setPitches(a, [C4+12]);
	print(getPitches(a));

	//print(randomInt(32));


	/* print(length(1)); length can only be called on arrays*/
}
