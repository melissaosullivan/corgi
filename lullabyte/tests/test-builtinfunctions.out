[A4, A5, B6, B7]
B5
0.5
5.
100
1
1
2
3
5.
50
[C#4, F5]
[C5]
